compiler/next/include/chpl/uast/VarArgFormal.h

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_VARARGFORMAL_H
+#define CHPL_UAST_VARARGFORMAL_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Formal.h"
+#include "chpl/uast/IntentList.h"
+#include "chpl/uast/VarLikeDecl.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a varargs formal.
+
+  \rst
+  .. code-block:: chapel
+      proc f(x: int ...?k) { writeln(x); }
+  \endrst
+
+  Here `x` is a formal that may take a variable number of actual arguments,
+  the number of which is denoted by the TypeQuery `?k`.
+*/
+class VarArgFormal final : public VarLikeDecl {
+ private:
+  int countChildNum_;
+
+  VarArgFormal(ASTList children, UniqueString name, Formal::Intent intent,
+               int8_t typeExpressionChildNum,
+               int8_t countChildNum)
+    : VarLikeDecl(asttags::VarArgFormal, std::move(children),
+                  Decl::DEFAULT_VISIBILITY,
+                  Decl::DEFAULT_LINKAGE,
+                  /*linkageNameChildNum*/ -1,
```

Comment:
Could you comment on what the -1 means in this context and the one below?

Reply:
When talking about child numbers (aka index into a vector of children in the uAST tree) we have been using -1 to indicate the absence of that child. (So here specifically it is saying that there is no linkage name child uAST node).